File: front/lib/data/models/community_model/place_review.dart
class PlaceReview {
  String? prNo;
  String? placeName;
  String? thumb;
  String? addressName;
  String? roadAddressName;
  String? userNo;
  String? contents;
  String? category;
  double? latitude; // 위도
  double? longitude; // 경도
  int? heart;

  String? userNick;
  String? imageUrl;

  PlaceReview(
      this.prNo,
      this.placeName,
      this.thumb,
      this.addressName,
      this.roadAddressName,
      this.userNo,
      this.contents,
      this.category,
      this.latitude,
      this.longitude,
      this.heart,
      this.userNick,
      this.imageUrl);

  PlaceReview.fromJson(Map<String, dynamic> json)
      : prNo = json['prNo'],
        placeName = json['placeName'],
        thumb = json['thumb'],
        addressName = json['addressName'],
        roadAddressName = json['roadAddressName'],
        userNo = json['userNo'],
        contents = json['contents'],
        category = json['category'],
        latitude = json['latitude'],
        longitude = json['longitude'],
        heart = json['heart'],
        userNick = json['userNick'],
        imageUrl = json['imageUrl'];

  Map<String, dynamic> toJson() {
    return {
      "prNo": prNo,
      "placeName": placeName,
      "thumb": thumb,
      "addressName": addressName,
      "roadAddressName": roadAddressName,
      "userNo": userNo,
      "contents": contents,
      "category": category,
      "latitude": latitude,
      "longitude": longitude,
      "heart": heart,
      "userNick": userNick,
      "imageUrl": imageUrl,
    };
  }

  @override
  String toString() {
    return 'PlaceReview{prNo: $prNo, placeName: $placeName, thumb: $thumb, addressName: $addressName, roadAddressName: $roadAddressName, userNo: $userNo, contents: $contents, category: $category, latitude: $latitude, longitude: $longitude, heart: $heart, userName: $userNick, imageUrl: $imageUrl}';
  }
}

File: back/src/main/java/com/pingo/entity/community/PlaceReview.java
package com.pingo.entity.community;

import com.pingo.exception.BusinessException;
import com.pingo.exception.ExceptionCode;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.ToString;

import java.util.UUID;

@Getter
@ToString
@AllArgsConstructor
@NoArgsConstructor
public class PlaceReview {
    private String prNo;
    private String placeName;
    private String thumb;
    private String addressName;
    private String roadAddressName;
    private String userNo;
    private String contents;
    private String category;
    private double latitude;
    private double longitude;
    private int heart;

    public void createPrNo() {
        if(this.prNo == null) {
            this.prNo = "PR" + createNo();
        }else {
            throw new BusinessException(ExceptionCode.DUPLICATE_PLACE_REVIEW_NO);
        }
    }

    public String createThumbName() {
        if(this.thumb == null) {
            return "PI" + createNo();
        }else {
            throw new BusinessException(ExceptionCode.DUPLICATE_REVIEW_IMAGE_NO);
        }
    }

    private String createNo() {
        return UUID.randomUUID().toString().substring(0, 8);
    }

    public void insertThumb(String imageUrl) {
        this.thumb = imageUrl;
    }
}
